compiler/z/codegen/OMRTreeEvaluatorTable.hpp

```diff
@@ -24,780 +24,774 @@
+#define datanZEvaluator TR::TreeEvaluator::libmFuncEvaluator
+#define datan2ZEvaluator TR::TreeEvaluator::libmFuncEvaluator
+#define dlogZEvaluator TR::TreeEvaluator::libmFuncEvaluator
+#define dfloorZEvaluator TR::TreeEvaluator::dfloorEvaluator
+#define ffloorZEvaluator TR::TreeEvaluator::ffloorEvaluator
+#define dceilZEvaluator TR::TreeEvaluator::dceilEvaluator
+#define fceilZEvaluator TR::TreeEvaluator::fceilEvaluator
+#define ibranchZEvaluator TR::TreeEvaluator::ibranchEvaluator
+#define mbranchZEvaluator TR::TreeEvaluator::mbranchEvaluator
+#define getpmZEvaluator TR::TreeEvaluator::getpmEvaluator
+#define setpmZEvaluator TR::TreeEvaluator::setpmEvaluator
+#define loadAutoOffsetZEvaluator TR::TreeEvaluator::loadAutoOffsetEvaluator
+#define imaxZEvaluator TR::TreeEvaluator::imaxEvaluator
+#define iumaxZEvaluator TR::TreeEvaluator::imaxEvaluator
+#define lmaxZEvaluator TR::TreeEvaluator::lmaxEvaluator
+#define lumaxZEvaluator TR::TreeEvaluator::lmaxEvaluator
+#define fmaxZEvaluator TR::TreeEvaluator::fmaxEvaluator
+#define dmaxZEvaluator TR::TreeEvaluator::dmaxEvaluator
+#define iminZEvaluator TR::TreeEvaluator::iminEvaluator
+#define iuminZEvaluator TR::TreeEvaluator::iminEvaluator
+#define lminZEvaluator TR::TreeEvaluator::lminEvaluator
+#define luminZEvaluator TR::TreeEvaluator::lminEvaluator
+#define fminZEvaluator TR::TreeEvaluator::fminEvaluator
+#define dminZEvaluator TR::TreeEvaluator::dminEvaluator
+#define trtZEvaluator TR::TreeEvaluator::trtEvaluator
+#define trtSimpleZEvaluator TR::TreeEvaluator::trtEvaluator
+#define ihbitZEvaluator TR::TreeEvaluator::integerHighestOneBit
+#define ilbitZEvaluator TR::TreeEvaluator::unImpOpEvaluator
+#define inolzZEvaluator TR::TreeEvaluator::integerNumberOfLeadingZeros
+#define inotzZEvaluator TR::TreeEvaluator::integerNumberOfTrailingZeros
+#define ipopcntZEvaluator TR::TreeEvaluator::integerBitCount
+#define lhbitZEvaluator TR::TreeEvaluator::longHighestOneBit
+#define llbitZEvaluator TR::TreeEvaluator::unImpOpEvaluator
+#define lnolzZEvaluator TR::TreeEvaluator::longNumberOfLeadingZeros
+#define lnotzZEvaluator TR::TreeEvaluator::longNumberOfTrailingZeros
+#define lpopcntZEvaluator TR::TreeEvaluator::longBitCount
+#define ibyteswapZEvaluator TR::TreeEvaluator::unImpOpEvaluator
+#define bbitpermuteZEvaluator TR::TreeEvaluator::bitpermuteEvaluator
+#define sbitpermuteZEvaluator TR::TreeEvaluator::bitpermuteEvaluator
+#define ibitpermuteZEvaluator TR::TreeEvaluator::bitpermuteEvaluator
+#define lbitpermuteZEvaluator TR::TreeEvaluator::bitpermuteEvaluator
+#define PrefetchZEvaluator TR::TreeEvaluator::PrefetchEvaluator
+
+
+
+#include "il/OMROpcodes.hpp"
+
+#define GENERATE_TREE_EVAL_TABLE(\
+   opcode, \
+   name, \
+   prop1, \
+   prop2, \
+   prop3, \
+   prop4, \
+   dataType, \
+   typeProps, \
+   childProps, \
+   swapChildrenOpcode, \
+   reverseBranchOpcode, \
+   boolCompareOpcode, \
+   ifCompareOpcode, \
+   enumValue, \
+   ...) enumValue ## ZEvaluator,
```

Comment:
Is there a functional reason why we had to include the arch in the name? It is less useful now that we have for example `iloadZEvaluator` and `iloadPowerEvaluator` as we can't search of all the evaluators on all platforms easily and compare them, which is something often done.

IMO the goal should be to be as similar across platforms as possible. The platform specific code is already in its own directory for example `/z/...` and `/p/...`.

Reply:
That's a good point. Should we discuss this more, or can I proceed with refactoring?

Reply:
Yes we should discuss. Let me tag a few people here @0xdaryl @aviansie-ben @Leonardo2718. Thoughts?

Reply:
I agree that including the arch in the evaluator name doesn't provide that much value. I think dropping it makes sense.

Reply:
@0xdaryl @aviansie-ben any other thoughts?

Reply:
I agree the architecture name shouldn't be there.

Reply:
I think we have enough of a consensus to revert back the names without the platform suffixes and make them uniform across codegens.

Reply:
It looks like there is a naming collision if I remove architecture name from the macro name, i.e., `#define iloadEvaluator TR::...::iloadEvaluator` probably re-defines (and re-expands) when a macro is called.

Do you think having underscores for the macro names would be OK? I.e., smth like `#define _iloadEvaluator TR...`

Reply:
That seems reasonable to me.